src/mapgen/mg_decoration.h

```diff
@@ -112,12 +113,15 @@ class DecoSchematic : public Decoration {
 };
 
 
-/*
 class DecoLSystem : public Decoration {
 public:
-	virtual void generate(Mapgen *mg, u32 blockseed, v3s16 nmin, v3s16 nmax);
+	ObjDef *clone() const;
+
+	virtual size_t generate(MMVManip *vm, PcgRandom *pr, v3s16 p, bool ceiling);
+
+	const NodeDefManager *ndef;
```

Comment:
also noticed something else, `Decoration` inherits `NodeResolver` so you already have `m_ndef`

Reply:
Now it reuses the `NodeDefManager` from `NodeResolver`.